test/utils.js

```diff
@@ -1,26 +1,46 @@
 const expect = require('chai').expect;
+const fs = require('fs')
+const sinon = require('sinon')
 module.exports = {
-    generateTests: (parser, TestCase) => {
-        const funs = Object.keys(TestCase);
-
+    generateTests: (parser, TestCase, type) => {
+        let clock = null;
+            now = new Date();
+        beforeEach(() => {
+            clock = sinon.useFakeTimers(now.getTime())
+        })
+        afterEach(() => {
+            clock.restore()
+        })
+        const funs = Object.keys(TestCase)
         funs.forEach(fun => {
             it(fun, () => {
-                const formulas = Object.keys(TestCase[fun]);
+                const formulas = Object.keys(TestCase[fun])
                 formulas.forEach(formula => {
-                    const expected = TestCase[fun][formula];
-                    let result = parser.parse(formula, {row: 1, col: 1});
+                    let expected = TestCase[fun][formula]
+                    let result = parser.parse(formula, {row:1, col:1});
                     if (result.result) result = result.result;
+                    //variable in nature, so I can not save them in a JSON
+                    if(formula === "HOUR(NOW())")
```

Comment:
Override the date function so it's deterministic.
https://stackoverflow.com/questions/1976968/change-browser-time-to-test-return-value-of-date